train_cnn.py

```diff
@@ -0,0 +1,147 @@
+"""
+CNN (encoder performing unsupervised feature learning by mapping an input image to a learned fixed-length vector):
+
+Image (625x1250)
+  --> 3x3 receptive fields convolved with stride 1 as in VGGNet (this done twice)
+    --> fixed-size output vector
+      --> CNN(width=32)
+        --> CNN(width=64)
+          --> CNN(width=128)
+            --> 2 fully-connected layers (size=1024) applying the rectified linear unit activation
+              --> output vector (to be concatenated with 1st LSTM output vector)
+
+"""
+
+import tensorflow as tf
+import numpy as np
+from helpers.utils import load_pickle
+from definitions import data_dir
+
+
+def extract_data(path):
+  print 'Unpickling {}'.format(path.split('/').pop())
+  
+  data = load_pickle(path)
+  
+  return data['data'], data['labels']
+
+
+def conv_2d(x, dims=(3, 3), filters=32, strides=(1, 1),
+           std=1e-3, padding='SAME', activation=tf.identity, scope='conv2d', reuse=False):
+  with tf.variable_scope(scope, reuse=reuse):
+    s = x.get_shape().as_list()
+    
+    shape = list(dims) + [s[3], filters]
+  
+    W = tf.Variable(tf.random_normal(shape=shape, stddev=std), name='W')
+    b = tf.Variable(tf.ones([filters]) * std, name='b')
+    o = tf.nn.convolution(x, W, padding, strides=strides)
+    o = o + b
+    
+    return activation(o)
+
+
+def fully_connected(x, output_units=100, activation=tf.identity, std=1e-3, scope='fc', reuse=False):
+  with tf.variable_scope(scope, reuse=reuse):
+    s = x.get_shape().as_list()
+    
+    shape = [s[1], output_units]
+    
+    W = tf.Variable(tf.random_normal(shape, stddev=std))
+    b = tf.Variable(tf.ones([shape[1]]) * std)
+    h = tf.matmul(x, W) + b
+
+    return activation(h)
+
+
+def get_batch(X, Y, N, n=32):
+  inds = np.random.choice(range(N), size=n, replace=False)
+  return X[inds], Y[inds]
+
+
+if __name__ == '__main__':
+  # Extract pickled train, validation, and test data
+  X_train, Y_train = extract_data('{}/train.pkl'.format(data_dir))
+  X_val, Y_val = extract_data('{}/validation.pkl'.format(data_dir))
+  X_test, Y_test = extract_data('{}/test.pkl'.format(data_dir))
```

Comment:
^Using a train/validation/test ratio of 0.6/0.2/0.2 right now, with only 10 total images just to get the hang of things.